classes/Task/Restore/Restore.php

```diff
@@ -49,14 +49,14 @@ class Restore extends AbstractTask
      */
     public function run(): int
     {
-        $this->container->getState()->setProgressPercentage(
+        $this->container->getRestoreState()->setProgressPercentage(
```

Comment:
I think you can variabilise the state again `$state = $this->container->getRestoreState()` instead of reusing all the time `$this->container->getRestoreState()`